packages/vow/test/retryable-restart.test.js

```diff
@@ -0,0 +1,110 @@
+import {
+  annihilate,
+  getBaggage,
+  nextCrank,
+  startLife,
+  test,
+} from '@agoric/swingset-vat/tools/prepare-strict-test-env-ava.js';
+
+import { Fail } from '@endo/errors';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { prepareVowTools } from '../vat.js';
+
+test.serial('retries on disconnection', async t => {
+  annihilate();
+
+  t.plan(1);
+
+  await startLife(
+    async baggage => {
+      const zone = makeDurableZone(baggage, 'durableRoot');
+      const { retryable, watch } = prepareVowTools(zone);
+      const retry = retryable(zone, 'retry', async () => {
+        // Never resolves, simulates external call
+        await new Promise(() => {});
+      });
+
+      const watcher = zone.exo('DurableVowTestWatcher', undefined, {
+        onFulfilled(value) {
+          t.fail(
+            `First incarnation watcher onFulfilled triggered with value ${value}`,
+          );
+        },
+        onRejected(reason) {
+          t.fail(
+            `First incarnation watcher onRejected triggered with reason ${reason}`,
+          );
+        },
+      });
+
+      return { zone, watch, retry, watcher };
+    },
+    async ({ zone, watch, retry, watcher }) => {
+      const result = retry();
+      zone.makeOnce('result', () => result);
+      watch(result, watcher);
+      await nextCrank();
+    },
+  );
+
+  await startLife(
+    baggage => {
+      const zone = makeDurableZone(baggage, 'durableRoot');
+      const { retryable, when } = prepareVowTools(zone);
+
+      // Reconnect retryable definition
+      retryable(zone, 'retry', async () => {
+        // Simulate call that settles
+        await nextCrank();
+        return 42;
+      });
+
+      zone.exo('DurableVowTestWatcher', undefined, {
+        onFulfilled(value) {
+          t.is(value, 42, 'vow resolved with value 42');
```

Comment:
consider losing the `t.plan` to make the logic clearer. E.g. in the test closer define a promise kit and resolve it with the `when(result)`. Then at the end of the test have `await t.is(await resultP, 24)`, something like that.

Not a biggy but would make the logic easier to read and modify

Reply:
A promise kit wouldn't help because it doesn't guarantee the other handlers didn't trigger (as promise resolve logic ignores any subsequent resolution). Of course we could set a counter to check how many times the handlers got triggered, but at that point, isn't that what `plan` does?

Reply:
> because it doesn't guarantee the other handlers didn't trigger

i was thinking
```suggestion
          if (value === 42) {
            p.resolve('success');
          }
```
But you're right something else could resolve with `success`, so I see now that `t.plan()` is safer